libsimulator/src/GCFMModel.cpp

```diff
@@ -95,13 +92,19 @@ void GCFMModel::CheckDistanceConstraint(
     const NeighborhoodSearchType& neighborhoodSearch) const
 {
     const auto neighbors = neighborhoodSearch.GetNeighboringAgents(agent.pos, 2);
+    const auto& model = std::get<VelocityModelData>(agent.model);
     for(const auto& neighbor : neighbors) {
-        const auto spacing = AgentToAgentSpacing(agent, neighbor);
-        if(spacing <= 0) {
+        const auto& neighborModel = std::get<VelocityModelData>(agent.model);
+        const auto contanctDist = model.radius + neighborModel.radius;
+        const auto distance = (agent.pos - neighbor.pos).Norm();
```

Comment:
Is there a reason for not using `AgentToAgentSpacing` anymore?

Reply:
good catch